### DISCUSSION TITLE
InfluxDB as Docker Container on another device?

### DISCUSSION THREAD

**Discussion by vulture2600** [David00/rpi-power-monitor#133] [General]
Hello,
Is it possible for the PowerMonitor service to write to an InfluxDB Docker Container? I've tried a bunch of things including starting a docker image of influxDB, installing influxDB into an Ubuntu container and a few other rabbit holes. Id like to have the Pi write to an InfluxDB database running on my NAS over the internet instead of to a local instance of InfluxDB running on the same Pi. Any ideas?
I tried editing the config.toml file to point to the docker container but no luck there. It seems like the config file should be able to point the data to wherever you want, as long as theres an influx server to write to, correct?
Thanks!

**Comment by David00** [comment]
Hi @vulture2600, yes, it is possible to write to a remote InfluxDB server, but as of now, only Influx v1.8 is supported. See here:
https://david00.github.io/rpi-power-monitor/docs/v0.3.0/advanced-usage.html#using-a-remote-influxdb-server

I have been wanting to bring support for Influx version 2 for awhile now as it has much better data processing capabilities. Would you be interested in assisting with testing if I add InfluxDB version 2 support?

**Reply by vulture2600** [reply]
Thank you! I'll see if I can figure out how to install v1.8 in a debian container. As far as testing goes, if you can steer me in the right direction I'd be willing to help. I know very little about  databases.

**Reply by mholzma** [reply]
I'd be happy to assist in testing InfluxDB version 2

**Comment by vulture2600** [comment]
I got it working, but for some reason theres now about 50 seconds of lag. I found that my Nas that hosts the docker container was not syncing to NTP, so I fixed that and now my docker containers sync to my Nas. That reduced the lag by 9 minutes. I'm still trying to figure out the remaining lag. I know have two databases on the InfluxDB server and two Pi's talking to them.
